spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -677,7 +677,13 @@ object QueryPlanSerde extends Logging with CometExprShim {
         }
         None
 
-      case div @ IntegralDivide(left, right, _) if supportedDataType(left.dataType) =>
```

Comment:
I thought that we needed to make this change for the `Divide` expression rather than the `IntegralDivide` expression?

Reply:
As discussed yesterday, the issue seems to be with IntegralDivide where in the current implementation in Spark is not in alignment with Spark's integral divide (where all the operations are converted to float division except decimal type).

Reply:
The fix essentially mimcs Spark's behavior to have divide operands being casted to DoubleType (except for Decimal type)